packages/manager/apps/hub/src/AutorenewService.class.js

```diff
@@ -0,0 +1,284 @@
+import assign from 'lodash/assign';
+import head from 'lodash/head';
+import includes from 'lodash/includes';
+import isNull from 'lodash/isNull';
+import snakeCase from 'lodash/snakeCase';
+import 'moment';
+
+import { DEBT_STATUS } from './billing-service.constants';
+
+export default class BillingService {
+  constructor(service) {
+    Object.assign(this, service);
+
+    this.expirationDate = moment(this.expiration);
+
+    if (this.status) {
+      this.state = this.isSuspended() ? 'EXPIRED' : 'UP';
+    }
+  }
+
+  getServiceName() {
+    return this.serviceId;
+  }
+
+  get formattedExpiration() {
+    return this.expirationDate.format('LL');
+  }
+
+  getRenew() {
+    if (this.hasManualRenew()) {
+      return 'manualPayment';
+    }
+
+    if (this.shouldDeleteAtExpiration() && !this.isResiliated()) {
+      return 'delete_at_expiration';
+    }
+
+    if (this.isResiliated()) {
+      return 'expired';
+    }
+
+    if (this.hasAutomaticRenew() || this.hasForcedRenew()) {
+      return 'automatic';
+    }
+
+    return 'manualPayment';
+  }
+
+  hasAutomaticRenew() {
+    return this.renew.automatic;
+  }
+
+  hasForcedRenew() {
+    return (
+      this.renew.forced && !this.shouldDeleteAtExpiration() && !this.isExpired()
```

Comment:
I dont think the 2 API `billingServices` returns `renew.forced` and many other required variables. Probably the 2API has to be updated to suit this class

Reply:
@ganeshkumar1989 a PR is pending on 2API